### ISSUE TITLE
GPS WNRO: Apr 7, 2019

### ISSUE THREAD

**Issue by tomkooij** [HiSPARC/datastore#31]
De dag die je wist dat zou komen...

Apr 7, 2019 is the second `GPS WNRO` (week number rollover). On this day the 10-bit GPS week number overflowed from 1023 back to 0. GPS week 0 started on Jan 6, 1980. This was the second rollover. The GPS weeknumber of Apr 7, 2019 is 2048.

Our Trimble Resolution T GPS clocks use the start of weeknumber 2048 (Apr 7, 2019 0:00) as the default time after a GPS cold start (no signal acquired). The DAQ will hapilly send events to the datastore evenwhen no GPS signal has been acquired. Events from Apr 7, 2019 are thus very suspicious. Now that Apr 7, 2019 has passed we have verified that the GPS still use Apr 7, 2019 as default time after cold start.

Apr 7, 2019 used to be in the future... Not anymore.

Soon, we must mark events from Apr 7, 2019 as suspicious and not import them into the raw datastore, because there were most probably caused by missing/bad GPS signal.

(Last week I already moved the "old" Apr, 2019 suspicious data in the raw datastore)

**Comment by 153957** [comment]
What is the earliest date the Trimble GPS clocks have shown? I can understand that if they were produced after Aug 1999 (previous rollover) they may have a minimum date set in the firmware, such that if the week number if prior to that it will know that it needs to use the Apr 2019 reference. However, this might mean that somewhere in the future it will reach a week number where that is no longer true the dates might return to 200[0-9] values.. How do the GPS receivers know they need to use the new reference?

**Comment by 153957** [comment]
Apparently the Resolution T modules that are used in HiSPARC electronics are upgradable: https://www.trimble.com/wnro/products/timing.aspx (Note; the old Thunderbolts are not).
It might be possible to check the firmware using the DSPMon utility..

**Comment by tomkooij** [comment]
@153957: Most stations use Resolution T with firmware (`Application version` in DSPMon) 1.20 this the most recent and WNRO compliant. 
Cold start results in the date flipping back to 1999.

The most recent stations use the *Resolution SMT 360* (Resolution T was discontinued in 2013) which flip back to Apr 7, 2019 when cold starting.

**Comment by tomkooij** [comment]
TL;DR. If updated to the lastest (1.20) firware our Trimble Resolution T GPSes are usable until Nov 2029.

Thanks to the `time-nuts` mailling list (and Google) I found the dates we are looking for:
https://trimble.app.box.com/s/lp6267igvdu8xwv43p55irf74a70u608

![image](https://user-images.githubusercontent.com/6316468/55703412-d3efe380-59d9-11e9-98b8-42319dc7e525.png)

**Comment by 153957** [comment]
Great find! good to know.
